src/detect-file-hash-common.c

```diff
@@ -230,26 +229,28 @@ static DetectFileHashData *DetectFileHashParse (const DetectEngineCtx *de_ctx,
         goto error;
     }
 
-    rule_filename = SCStrdup(de_ctx->rule_file);
-    if (rule_filename == NULL) {
-        goto error;
-    }
-
     char line[8192] = "";
     fp = fopen(filename, "r");
     if (fp == NULL) {
 #ifdef HAVE_LIBGEN_H
         if (de_ctx->rule_file != NULL) {
+            char *rule_filename = SCStrdup(de_ctx->rule_file);
```

Comment:
Commit message says

> Move the copy into the HAVE_LIBGEN_H block

This is not important, it is rather the `if (de_ctx->rule_file != NULL)` block, and furthermore it is just restricting the scope f the variable to block where it is actually used instead of having the whole function as a scope